service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsHttpEndpoint.java

```diff
@@ -140,11 +141,11 @@ private void updateCredentials(final RoutingContext ctx) {
 
         // auth-id and type from URI must match payload
         if (!authIdFromUri.equals(authId)) {
-            ctx.response().setStatusMessage("Non-matching authentication identifier");
-            ctx.fail(HttpURLConnection.HTTP_BAD_REQUEST);
+            ctx.fail(new ServiceInvocationException(HttpURLConnection.HTTP_BAD_REQUEST,
```

Comment:
well, since 400 indicates a client error, we should create a `ClientErrorException` instead of a (generic) `ServiceInvocationException`
the same applies to multiple other locations ...

Reply:
I replaced it with `ClientErrorException`